app/models/document.rb

```diff
@@ -28,6 +32,19 @@ class Document < ApplicationRecord
   scope :of_type, ->(type) { where(document_type: type) }
 
   belongs_to :intake, optional: true
+  belongs_to :client, optional: true
   belongs_to :documents_request, optional: true
   has_one_attached :upload
+
+  before_save :set_display_name
+
+  def set_display_name
+    return if display_name
+
+    if upload.present?
+      self.display_name = upload.attachment.filename
```

Comment:
Curious why we're not modifying the filename directly. https://stackoverflow.com/a/52957480 should do the trick for that. Non-blocking.

Reply:
we talked about this with ben! decided it made sense to preserve the original filename